programs/utilities/spellbook.sc

```diff
@@ -0,0 +1,526 @@
+// You can continue adding spells to existing spellbooks. 
+// `/spellbook cat_spells set "spawn Tabby" /summon cat ~ ~ ~ {CatType:2}`
+// You can give yourself a copy of the spellbook with the spellbook give command,
+// `/spellbook cat_spells give`
+// Any changes you make to your spells will automatically update your spellbook when opened.
+// You can find more info in-game with the help command',
+// `/spellbook help <page>',
+// Pages: (main), basics, shorthands, customize, commands
+
+
+__config()->{
+  'command_permission' -> 'ops', 
+  'commands' -> {
+    '' -> ['help', 'main'],
+    'help' -> ['help', 'main'],
+    'help <helpPage>' -> 'help',
+    'list' -> 'list_books',
+    '<book> give' -> 'give_book',
+    '<book> delete' -> 'delete_book',
+    '<book> set <spell> <command>' -> 'set_command',
+    '<book> warp <spell>' -> 'set_warp_at_player',
+    '<book> warp <spell> at <location> in <dimension>' -> 'set_warp',
+    '<book> forceload <spell> <from> <to>' -> 'set_forceload_at_player',
+    '<book> forceload <spell> <from> <to> in <dimension>' -> 'set_forceload',
+    '<book> bot <spell> <bot>' -> 'set_bot_at_player',
+    '<book> bot <spell> <bot> at <location> in <dimension>' -> 'set_bot',
+    '<book> remove <spell>' -> 'delete_command',
+    '<book> read' -> ['display_book', 1],
+    '<book> read <spellPage>' -> 'display_book',
+    '<book> color <spell> <color>' -> 'set_spell_color',
+    '<book> tooltip <spell> <tooltip>' -> 'set_spell_tooltip'
+  },
+  'arguments' -> {
+    'command' -> {'type' -> 'text', 'suggest' -> ['/tp @p x y z', '/gamerule doFireTick true']},
+    'spell' -> {'type' -> 'string', 'suggest' -> ['"Which Farm Bot"', '"Warp to Spawn"', '"Fire Tick True"']},
+    'helpPage' -> {'type' -> 'term', 'options' -> ['main', 'basics', 'shorthands', 'customize', 'commands']},
+    'spellPage' -> {'type' -> 'int', 'suggest' -> [1,2,3,5]},
+    'tooltip' -> {'type' -> 'text', 'suggest' -> ['Spawn a witch', 'Turn Fire Tick on']},
+    'color' -> {'type' -> 'string', 'suggest' -> [
+      'dark_red', 'red', 'gold', 'yellow', 'dark_green', 'green', 'aqua', 'dark_aqua', 'dark_blue', 
+      'blue', 'light_purple', 'dark_purple', 'white', 'gray', 'dark_gray', 'black', '"#000000"', '"#ffffff"'
+    ]},
+    'book' -> {'type' -> 'string', 'suggester' -> _(args) -> (
+      options = ['bots', 'rules', 'warps'];
+      put(options, -1, _get_book_list(), 'extend');
+      options;
+    )},
+    'from' -> {'type' -> 'columnpos'},
+    'to' -> {'type' -> 'columnpos'},
+    'bot' -> {'type' -> 'term', 'suggest'-> ['Alex', 'Steve']},
+    'location' -> {'type' -> 'location'},
+    'dimension' -> {'type' -> 'dimension'}
+  }
+};
+
+
+global_help_pages = {
+  'main'-> [
+    'pb \nSpellBook\n', 
+    'm A utility used to create command books (spellbooks).\n',
+    'm Learn the basics with: ',
+    'ci /spellbook help basics\n',
+    'm Pages: ','ci (main), basics, shorthands, customize, commands'
```

Comment:
make these two be hyperkins? Click the name and have the help page in question showed